### ISSUE TITLE
AsyncServer fails to retain connection with clients

### ISSUE THREAD

**Issue by secretlyvogon** [miguelgrinberg/python-socketio#462]
From #324 

> On the server-side, I have defined an async coroutine, which communicates with the connected nodes. The clients receive the data, perform their functions, send an emit back to the server, and wait. The server collects data sent by each client. Once all the emits are collected, the server calls that coroutine again. Thus, forming a loop.
> 
> The first iteration of this loop executes properly, and the updates are collected as expected. However, right when the coroutine is called, the code seems to act oddly at the server end. By the time the coroutine emits, the clients are disconnected, and the following appears on the server-side log:
> 
`Client is gone, closing socket`

As per @miguelgrinberg suggestion, I tried experimenting with AsyncClient and regular Client with my AsyncServer. Following are the logs from both:

Async Client:

```
Sending packet PING data None
PONG response has not been received, aborting
Unexpected error "WebSocket read returned None", aborting
Waiting for write loop task to end
Exiting ping task
Exiting write loop task
Waiting for ping loop task to end
Engine.IO connection dropped
Exiting read loop task
Connection failed, new attempt in 0.70 seconds
```

Client:

```
Sending packet PING data None
PONG response has not been received, aborting
Exiting ping task
Exiting write loop task
WebSocket connection was closed, aborting
Waiting for write loop task to end
Waiting for ping loop task to end
Engine.IO connection dropped
Connection failed, new attempt in 1.16 seconds
Exiting read loop task
```

Looking at both the logs above it is certain that the server is having some issues in execution.

**Comment by miguelgrinberg** [comment]
As I said before, the most likely cause of this issue is that your server is blocking for long periods of time, which prevents the background tasks that keep all the connections with clients alive from running. I don't know how I can help with this, other than suggesting you look for places where you use blocking code that is incompatible with the asyncio concurrency model.

**Comment by secretlyvogon** [comment]
Okay, I am currently debugging it. Would like to keep this open for when I find the solution or if I need more help.

**Comment by secretlyvogon** [comment]
I found the error. I was using a python [websocket_client](https://pypi.org/project/websocket_client/) call on the server right before I called the async routine. That call was supposed to simply send data once and close the connection. I guess it closed all existing WebSocket connections or there was some sort of clash because of which all client connections dropped too.

Thank you for pointing me in the right direction, it saved me a lot of time in debugging. Closing the issue now.
